ethereum/contracts.go

```diff
@@ -22,4 +22,50 @@ type BridgeContract interface {
 		recipient []common.Address,
 		token []common.Address,
 	) (AssetsLockedIterator, error)
+
+	PastAssetsUnlockConfirmedEvents(
+		startBlock uint64,
+		endBlock *uint64,
+		unlockSequenceNumberFilter []*big.Int,
+		recipientFilter [][]byte,
+		tokenFilter []common.Address,
+	) ([]*MezoBridgeAssetsUnlockConfirmed, error)
+
+	ConfirmedUnlocks(arg0 *big.Int) (bool, error)
+}
+
+type MezoBridgeAssetsUnlockConfirmed struct { // TODO: Remove once bindings for `MezoBridge` are re-generated and contain features related to attestation.
+	UnlockSequenceNumber *big.Int
+	Recipient            common.Hash
+	Token                common.Address
+	Amount               *big.Int
+	Chain                uint8
+	Raw                  Log
+}
+
+type Log struct { // TODO: Remove once bindings for `MezoBridge` are re-generated and contain features related to attestation.
+	// Consensus fields:
+	// address of the contract that generated the event
+	Address common.Address `json:"address" gencodec:"required"`
+	// list of topics provided by the contract.
+	Topics []common.Hash `json:"topics" gencodec:"required"`
+	// supplied by the contract, usually ABI-encoded
+	Data []byte `json:"data" gencodec:"required"`
+
+	// Derived fields. These fields are filled in by the node
+	// but not secured by consensus.
+	// block in which the transaction was included
+	BlockNumber uint64 `json:"blockNumber" rlp:"-"`
+	// hash of the transaction
+	TxHash common.Hash `json:"transactionHash" gencodec:"required" rlp:"-"`
+	// index of the transaction in the block
+	TxIndex uint `json:"transactionIndex" rlp:"-"`
+	// hash of the block in which the transaction was included
+	BlockHash common.Hash `json:"blockHash" rlp:"-"`
+	// index of the log in the block
+	Index uint `json:"logIndex" rlp:"-"`
+
+	// The Removed field is true if this log was reverted due to a chain reorganization.
+	// You must pay attention to this field if you receive logs through a filter query.
+	Removed bool `json:"removed" rlp:"-"`
 }
```

Comment:
Do we really need to map the entire `Raw Log` field? We probably need the content of the AssetsUnlocked and maybe some block/time info, not the entire struct.

Reply:
This whole structure is not really needed and will be removed once we generate bindings for `MezoBridgeV2`.
I simply compiled MezoBridgeV2 locally and copied the whole structure from the artifacts to make everything compile. I suggest we keep the copied code here unchanged and remove it once the bindings are generated.
(There is a TODO for this).

Reply:
Ok, thanks for clarifying.